### PR TITLE
Blitz recipe-install support for NextJS App Router projects

### PR DESCRIPTION
**Pull request by Doc0x1**
## What are the changes and their implications?

Bringing the Blitz recipe-install feature up to speed with the newer NextJS App Router projects. 

TODO:

- [x] Change paths.ts so that it can check and return proper App Router paths/files. There's no _app file in App Router projects so as of the initial commit, it is set to return the root layout file of the project.
- [ ] Add in error handling for cases where folders may not exist in a NextJS App Router filesystem

Closes: #4189 

## Feature Checklist
- [x] Changeset added (run pnpm changeset in the root directory)
- [ ]  Documentation added/updated (submit PR to [blitzjs.com repo main branch](https://github.com/blitz-js/blitzjs.com))

---

Just a heads up, this is my first time contributing to a community project on GitHub, and so there's been a lot of new things I've been picking up in a pretty short amount of time. I've done my best to follow the contributing guidelines and if I've made any mistakes, feel free to let me know so I can avoid them in the future.

Anyway, this pull request is by no means ready for a merge, but I wanted to open it so I could get some feedback from those of you that might be more familiar with the recipe system than I currently am.

### CODE REVIEW

**Comment by Doc0x1** [comment]
I think the logical next step would be adding better error handling to prevent it from hanging during a recipe install. Having it create non-existent folders is one idea I've had.

**Comment by siddhsuresh** [comment]
> I think the logical next step would be adding better error handling to prevent it from hanging during a recipe install. Having it create non-existent folders is one idea I've had.
> 
> Looking over it, the issue happens at the .addTransformFilesStep part of the recipe, at least for the tailwind recipe. Working on a way to implement better error handling in the transform.ts file so it doesn't hang.

yeah @Doc0x1 that's a good idea. Let me know when the PR is ready for a review!

**Comment by Doc0x1** [comment]
> > I think the logical next step would be adding better error handling to prevent it from hanging during a recipe install. Having it create non-existent folders is one idea I've had.
> > 
> > Looking over it, the issue happens at the .addTransformFilesStep part of the recipe, at least for the tailwind recipe. Working on a way to implement better error handling in the transform.ts file so it doesn't hang.
> 
> yeah @Doc0x1 that's a good idea. Let me know when the PR is ready for a review!

Will do. I just got some good web dev work so its gonna be maybe a couple weeks before im able to get back to actively working on this, so just a heads up. Glad i could help with getting that other PR done though

File: recipes/tailwind/templates/config/tailwind.config.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // tailwind.config.js
 module.exports = {
-  content: ["./{src,app,pages}/**/*.{js,ts,jsx,tsx}"],
+  content: [
+    "./{src/pages,src/components,src/app,src,pages,components,app}/**/*.{js,ts,jsx,tsx,mdx}",
+  ],
```
Comment (siddhsuresh) [inline recipes/tailwind/templates/config/tailwind.config.js:5]: 🍰 https://nextjs.org/docs/app/building-your-application/styling/tailwind-css#configuring-tailwind
```suggestion
  content: [
    './app/**/*.{js,ts,jsx,tsx,mdx}',
    './pages/**/*.{js,ts,jsx,tsx,mdx}',
    './components/**/*.{js,ts,jsx,tsx,mdx}',
    './src/**/*.{js,ts,jsx,tsx,mdx}',
  ],
```

**Comment by justinbalaguer** [comment]
up

**Comment by mvarchdev** [comment]
I do not know what is the state, but I tried blitz and I have just issues, this is one of them... `ERR_INVALID_ARG_TYPE, The "path" argument must be of type string or an instance of Buffer or URL. Received undefined`, probably does soemthing with this

**Comment by JWBrownie** [comment]
I am still having this issue as of for version 2.0.10 is it safe to use the recipe even if it errors out?

Thanks,

**Comment by Doc0x1** [comment]
Might need to close this one, was working on updating it for the current upstream main branch of Blitz and not sure if I did so correctly.

**Comment by siddhsuresh** [comment]
Hey, I am closing this for now, feel free to reopen it and request a re-review when possible. Thanks!

**Pull request closed without merging**